commands/create_cluster_test.go

```diff
@@ -128,106 +179,114 @@ func Test_CreateClusterSuccessfully(t *testing.T) {
+			inputArgs: &addClusterArguments{
+				owner: "non-existing-owner",
+				token: "some-token",
+			},
+			serverResponseJSON: []byte(`{"code": "RESOURCE_NOT_FOUND", "message": "Lorem ipsum"}`),
+			responseStatus:     http.StatusNotFound,
+			errorMatcher:       IsOrganizationNotFoundError,
 		},
-		// extensive arguments (only a server error should let this fail)
 		{
-			apiEndpoint:         mockServer.URL,
-			clusterName:         "UnitTestCluster",
-			numWorkers:          4,
-			releaseVersion:      "0.3.0",
-			owner:               "acme",
-			token:               "fake token",
-			workerNumCPUs:       3,
-			workerMemorySizeGB:  4,
-			workerStorageSizeGB: 10,
-			verbose:             true,
+			description: "Server complains about a bad request",
+			inputArgs: &addClusterArguments{
+				owner:          "owner",
+				token:          "some-token",
+				releaseVersion: "100.200.300",
+			},
+			serverResponseJSON: []byte(`{"code": "INVALID_INPUT", "message": "Lorem ipsum"}`),
+			responseStatus:     http.StatusBadRequest,
+			errorMatcher:       IsBadRequestError,
 		},
-		// file not readable
 		{
-			apiEndpoint:   mockServer.URL,
-			token:         "fake token",
-			inputYAMLFile: "does/not/exist.yaml",
-			dryRun:        true,
+			description: "Non-existing YAML definition path",
+			inputArgs: &addClusterArguments{
+				owner:         "owner",
+				token:         "some-token",
+				inputYAMLFile: "does/not/exist.yaml",
+				dryRun:        true,
+			},
+			serverResponseJSON: []byte(``),
+			responseStatus:     0,
+			errorMatcher:       IsYAMLFileNotReadableError,
 		},
 	}
 
 	for i, testCase := range testCases {
-		validateErr := validateCreateClusterPreConditions(testCase)
-		if validateErr == nil {
-			_, execErr := addCluster(testCase)
-			if execErr == nil {
-				t.Errorf("Expected errors didn't occur in testCase %v", i)
+		t.Logf("Case %d: %s", i, testCase.description)
+
+		// mock server
+		mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			//t.Log("mockServer request: ", r.Method, r.URL)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(testCase.responseStatus)
+			w.Write([]byte(testCase.serverResponseJSON))
+		}))
+
+		// client
+		cmdAPIEndpoint = mockServer.URL // required to make initClient() work
+		testCase.inputArgs.apiEndpoint = mockServer.URL
+		err := initClient()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		validateErr := validateCreateClusterPreConditions(*testCase.inputArgs)
+		if validateErr != nil {
+			t.Errorf("Unexpected error in argument validation: %#v", validateErr)
+		} else {
+			_, err := addCluster(*testCase.inputArgs)
+			if err == nil {
+				t.Errorf("Test case %d did not yield an execution error.", i)
+			}
+			origErr := microerror.Cause(err)
+			if testCase.errorMatcher(origErr) == false {
+				t.Errorf("Test case %d did not yield the expected execution error, instead: %#v", i, err)
 			}
 		}
+
+		mockServer.Close()
```

Comment:
I understand refactoring new tests is a chore, but when you create a new test it would be good to comply with https://github.com/giantswarm/fmt/blob/master/go/unit_tests.md

Reply:
One of the changes actually brought one table test much closer to that proposed format.